packages/core/src/components/search/Filter/FilterDesktop.tsx

```diff
@@ -1,42 +1,46 @@
 import { setFacet, toggleFacet, useSearch } from '@faststore/sdk'
 
 import {
+  Button as UIButton,
   Filter as UIFilter,
   FilterFacetBoolean as UIFilterFacetBoolean,
   FilterFacetBooleanItem as UIFilterFacetBooleanItem,
   FilterFacetRange as UIFilterFacetRange,
   FilterFacets as UIFilterFacets,
+  Icon as UIIcon,
 } from '@faststore/ui'
 import { gql } from '@generated/gql'
-import type { Filter_FacetsFragment } from '@generated/graphql'
+import { deliveryPromise } from 'discovery.config.default'
 import { useFormattedPrice } from 'src/sdk/product/useFormattedPrice'
 import type { useFilter } from 'src/sdk/search/useFilter'
+import { FilterDeliveryOption } from './FilterDeliveryOption'
+import type { FilterSliderProps } from './FilterSlider'
 
-interface Props {
-  /**
-   * The array that represents the details of every facet.
-   */
-  facets: Filter_FacetsFragment[]
-  /**
-   * ID to find this component in testing tools (e.g.: cypress,
-   * testing-library, and jest).
-   */
-  testId?: string
-  /**
-   * Title for the `FilterDesktop` component.
-   */
-  title?: string
-}
+import { sessionStore } from 'src/sdk/session'
+
+interface FilterDesktopProps
+  extends Omit<
+    FilterSliderProps,
+    'onClose' | 'size' | 'direction' | 'applyBtnProps' | 'clearBtnProps'
+  > {}
 
 function FilterDesktop({
   facets,
   testId,
   dispatch,
   expanded,
   title,
-}: Props & ReturnType<typeof useFilter>) {
+  deliverySettings,
+}: FilterDesktopProps & ReturnType<typeof useFilter>) {
   const { resetInfiniteScroll, state, setState } = useSearch()
 
+  const shippingLabel = deliverySettings?.sectionTitle ?? 'Delivery'
+  const { postalCode } = sessionStore.read()
+
+  const filteredFacets = deliveryPromise.enabled
```

Comment:
I chose to apply this filter here because I thought it would be safer in case the delivery promise is not enabled in discovery.config and the response includes shipping facets. I wanted to keep the changes from `resolvers/facet.ts`